docs/common/install-best-practices.md

```diff
@@ -0,0 +1,139 @@
+Considerations for Large Sites
+------------------------------
+
+While `yum` is a wonderful tool for installing software on a single server, it's a poor tool to install the same version of the software on many hosts. We **strongly** recommend a cluster management tool; as far as we know, all cluster management tools provide a mechanism to create a local yum repository and have all your worker nodes use that.
+
+If you have more than 20 worker nodes, we have the following advice:
+
+-   Do **not** use one of the OSG repositories directly for worker node installations; build a local mirror instead. (See below).
+-   Distribute CRLs to the worker nodes using an HTTP proxy.
+
+Both items are covered in this document.
+
+In the future, we will be providing a mechanism for distributing CAs and CRLs via a shared file system; this is not quite finished. If you choose to do this, remember that your security infrastructure will only be as safe and reliable as the shared filesystem!
+
+Repository Mirrors
+------------------
+
+A local yum mirror allows you to reduce the amount of external bandwidth used when updating or installing packages.
+
+Add the following to a file in `/etc/cron.d`:
+
+``` screen
+%RED%RANDOM%ENDCOLOR% * * * * root rsync -aH rsync://repo.grid.iu.edu/osg/ /var/www/html/osg/
+```
+
+Or, to mirror only a single repository:
+
+``` screen
+%RED%RANDOM%ENDCOLOR% * * * * root rsync -aH rsync://repo.grid.iu.edu/osg/3.1/el6/development /var/www/html/osg/3.1/el6
+```
+
+Replace %RED%RANDOM<span class="twiki-macro ENDCOLOR"></span> with a number between 0 and 59.
+
+On your worker node, you can replace the `baseurl` line of `/etc/yum.repos.d/osg.repo` with the appropriate URL for your mirror.
+
+If your osg\*.repo files still point to the legacy 3.0 repo layout (eg, /3.0/el5/osg-release/ instead of the new /osg/3.1/el5/release/), they can be updated to point to the new 3.1 layout by installing the latest version of the 'osg-release' package:
+
+``` screen
+yum update --enablerepo=osg-testing osg-release
+```
+
+If you are interested in having your mirror be part of the OSG's default set of mirrors, [please file a GOC ticket](https://ticket.grid.iu.edu/).
+
+CA Certificate Installation Considerations
+------------------------------------------
+
+CAs are distributed in two ways:
+
+-   As an RPM that contains the set of CAs. There are several such RPMs corresponding to different sets of CAs.
+-   Through direct downloads from GOC with the `osg-update-certs` tool (provided by the `osg-ca-scripts` RPM).
+
+As long as you use one of these two mechanisms, the OSG software will install successfully.
+
+Certificate Revocation List (CRL) Installation/Update
+-----------------------------------------------------
+
+CRLs are not distributed via `yum`. Instead, we provide the `fetch-crl` tool that downloads CRLs to the CA directory.
+
+<span class="twiki-macro INCLUDE" section="FetchCRLConfigProxy">InstallCertAuth</span>
+
+Removing Large numbers of RPMs
```

Comment:
Drop this entire section:
- We don't support RHEL 5 anymore
- We don't ship yum-remove-osg anymore
- The rest of it isn't very useful